File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/mediator/lab/SetupClass.java
// SetupClass.java

public class SetupClass
{
	public static Consumer setup()
	{
		Producer aProducer1 = new Producer("Alpha");
		Producer aProducer2 = new Producer("Bravo");
		Producer aProducer3 = new Producer("Charlie");
		Producer aProducer4 = new Producer("Delta");
		Producer aProducer5 = new Producer("Echo");
		Producer aProducer6 = new Producer("Foxtrot");

		Consumer aConsumer = new Consumer("Zulu");
		aConsumer.addProducer(aProducer1);
		aConsumer.addProducer(aProducer2);
		aConsumer.addProducer(aProducer3);
		aConsumer.addProducer(aProducer4);
		aConsumer.addProducer(aProducer5);
		aConsumer.addProducer(aProducer6);

		return aConsumer;
	}
}

File: dp_labs_and_solutions/dp_labs_and_solutions/dp_matlab_labs_and_solutions/mediator/lab/SetupClass.m
% SetupClass.m

classdef SetupClass < handle
	methods (Static)
		function result = setup
			aProducer1 = Producer('Alpha');
			aProducer2 = Producer('Bravo');
			aProducer3 = Producer('Charlie');
			aProducer4 = Producer('Delta');
			aProducer5 = Producer('Echo');
			aProducer6 = Producer('Foxtrot');

			aConsumer = Consumer('Zulu');
			aConsumer.addProducer(aProducer1);
			aConsumer.addProducer(aProducer2);
			aConsumer.addProducer(aProducer3);
			aConsumer.addProducer(aProducer4);
			aConsumer.addProducer(aProducer5);
			aConsumer.addProducer(aProducer6);

			result = aConsumer;
		end
	end
end
